src/app/operators/pipes/filter.pipe.ts

```diff
@@ -0,0 +1,13 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'filter'
+})
+export class FilterPipe implements PipeTransform {
+  transform(items: any[], filter: string): any {
+    if (!items || !filter) {
+      return items;
+    }
+    return items.filter(item => item.name.indexOf(filter) !== -1);
```

Comment:
I don´t like the pipe approach for this case.

Having no type is a two edged blade: doesn't tight the use of the pipe with a concrete type, but you lose type check at compile time.